### PR TITLE
fix: hyprlands window-rewrite 

### PR DESCRIPTION
**Pull request by khaneliman**
Resolves: https://github.com/Alexays/Waybar/issues/2718 found during testing another feature in pipeline. The window rewrites work on boot for me and no issues dragging between monitors anymore. 

cc @Syndelis

### CODE REVIEW

File: src/modules/hyprland/workspaces.cpp [hunk 1]
```diff
@@ -181,6 +181,9 @@ auto Workspaces::update() -> void {
   }
 
   for (auto &workspace : m_workspaces) {
+    auto clients_data = gIPC->getSocket1JsonReply("clients");
```
Comment (zjeffer) [inline src/modules/hyprland/workspaces.cpp:184]: Shouldn't you get the clients_data outside of the for-loop? Right now you're getting the data every iteration of the loop, does the data change in between iterations?

Comment (khaneliman) [reply src/modules/hyprland/workspaces.cpp]: Good catch, pasted it inside loop accidentally.

**Comment by Syndelis** [comment]
Although this would solve the issue, I feel like getting the `clients` information for every update seems overkill. The problem seems to appear when a window is dragged to a workspace that doens't exist in tha same bar it originated from. A less performance-intensive solution would be to check if the icon is empty when the window is moved to another workspace [here](https://github.com/Syndelis/Waybar/blob/8b1c0f3f7aee14b01eca6c010c157dc13e0c5b81/src/modules/hyprland/workspaces.cpp#L401-L407). I'll see what I can hack together tomorrow and will get back here.

**Comment by khaneliman** [comment]
Oh yeah, it's definitely not the most elegant... I was just excited to finally see the bar function the way I had hoped finally..

**Comment by Syndelis** [comment]
Sorry for being late, @khaneliman ! I've opened #2817 which should hopefully fix this problem as well. Could you verify that works for your setup?

**Comment by khaneliman** [comment]
closing with work being done in https://github.com/Alexays/Waybar/pull/2817

**Pull request closed without merging**